### ISSUE TITLE
internal error from `var a = nil;`

### ISSUE THREAD

**Issue by cassella** [chapel-lang/chapel#15136]
### Steps to Reproduce

**Source Code:**

```chapel
var a = nil;
```

```
foo.chpl:1: internal error: no field 'nil' in class 'heap__nilType' in getField() [AST/AggregateType.cpp:1967]
```

or without developer mode,

```
foo.chpl:1: internal error: AST-AGG-YPE-1967 chpl version 1.21.0 pre-release (2c9530a)
```

**Associated Future Test(s):**

None yet

### Configuration Information

```
chpl version 1.21.0 pre-release (2c9530a)

CHPL_TARGET_PLATFORM: linux64
CHPL_TARGET_COMPILER: gnu
CHPL_TARGET_ARCH: x86_64
CHPL_TARGET_CPU: native *
CHPL_LOCALE_MODEL: flat
CHPL_COMM: gasnet *
  CHPL_COMM_SUBSTRATE: udp
  CHPL_GASNET_SEGMENT: everything
CHPL_TASKS: qthreads
CHPL_LAUNCHER: amudprun
CHPL_TIMERS: generic
CHPL_UNWIND: none
CHPL_MEM: jemalloc
CHPL_ATOMICS: cstdlib
  CHPL_NETWORK_ATOMICS: none
CHPL_GMP: gmp
CHPL_HWLOC: hwloc
CHPL_REGEXP: re2
CHPL_LLVM: none
CHPL_AUX_FILESYS: none

gcc (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
```

**Comment by bradcray** [comment]
Well that's a bit sad.  Thanks for reporting this, Paul!

FWIW, this program compiled and ran under 1.20, though the output seems a little questionable to me:  [TIO](https://tio.run/##S85ILEjN@f@/LLFIIVHBViEvM8eaq7wosyQ1J08jUROJrVdSWZBqVVxSlJmXrmn9/z8A)

```chapel
var a = nil;
writeln(a);
writeln(a.type:string);
```

```
0x0
nil
```

**Comment by cassella** [comment]
This was fixed by #19672, which added an equivalent test.
